### PR TITLE
Fix long number formats

### LINKED ISSUE #562: [D3FC] - Charts do not handle large numerical values on the Y-axis
### Steps to Reproduce:
 1. Include a dataset that contains very large numerical values (e.g. 100,000,000+)
 2. Select a D3FC chart (e.g. d3fc Y Line).
 3. Add the large value to the chart

### Expected Result:
The values should be displayed and visible on the Y-axis

### Actual Result:
Only the last 8 digits of the numbers are displayed:

**D3FC**
image

### Highcharts equivalent
In the equivalent Highcharts implementation, large numbers are formatted as 100M (for million), this prevents the labels getting too big:

**Highcharts**
image

### PR DESCRIPTION
**Pull request by mhooper-scottlogic**
Fix #562 by creating a valueFormatter that displays very long numbers in SI format

### CODE REVIEW

**Review by andy-lee-eng — CHANGES_REQUESTED** [review]
How about if `axis/linearAxis.js` exports a "format" function (the others can leave it undefined). Then in `axisFactory` you can just pass the "format" function through (the same as we do the "decorate" function). Finally, in `chartFactory`, you can apply the "format" function to the correct axis (again the same as for "decorate").

I don't think you should need to modify all the charts for this.

File: packages/perspective-viewer-d3fc/src/js/axis/linearAxis.js [hunk 1]
```diff
@@ -68,3 +69,5 @@ export const domain = () => {
 };
 
 export const labelFunction = valueName => d => d[valueName][0];
+
+export const tickFormatFunction = d => (Math.abs(d) >= 10000000 ? d3.format(".3s")(d) : Number.isInteger(d) ? d3.format(",.0f")(d) : d3.format(",.2f")(d));
```
Comment (andy-lee-eng) [inline packages/perspective-viewer-d3fc/src/js/axis/linearAxis.js]: Should it use `valueformatter`?

File: packages/perspective-viewer-d3fc/src/js/axis/axisFactory.js [hunk 2]
```diff
@@ -58,7 +58,9 @@ export const axisFactory = settings => {
             },
             size: component.size,
             decorate: component.decorate,
-            label: settings[settingName].map(v => v.name).join(", ")
+            label: settings[settingName].map(v => v.name).join(", "),
+            type: useType,
```
Comment (andy-lee-eng) [inline packages/perspective-viewer-d3fc/src/js/axis/axisFactory.js]: Do you still need to include `type`?

**Review by andy-lee-eng — CHANGES_REQUESTED** [review]

File: packages/perspective-viewer-d3fc/src/js/axis/linearAxis.js (hunk 1 above)
Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/axis/linearAxis.js]: Yes, I have switched to use valueformatter.

File: packages/perspective-viewer-d3fc/src/js/axis/axisFactory.js (hunk 2 above)
Comment (mhooper-scottlogic) [reply packages/perspective-viewer-d3fc/src/js/axis/axisFactory.js]: type removed

**Review by andy-lee-eng — APPROVED** [review]

**Pull request merged**